File: sistema_de_gerenciamento.c
typedef struct {
    char nome[50];
    int id;
    char localizacao[50];
} Departamento;

typedef struct {
    char nome[50];
    int id;
    char cargo[50];
    float salario;
    Departamento departamento;
} Empregado;

void atualizar_informacoes(Empregado *empregado, char novo_nome[50], char novo_cargo[50], float novo_salario);
void transferir_departamento(Empregado *empregado, Departamento novo_departamento);
Departamento obter_info_departamento(Empregado *empregado);

void adicionar_empregado(Departamento *departamento, Empregado novo_empregado);
void remover_empregado(Departamento *departamento, int id_empregado);
void obter_info_empregados(Departamento *departamento);

File: sistema_de_gerenciamento.java
public class Departamento {
    private String nome;
    private int id;
    private String localizacao;

    public void adicionarEmpregado(Empregado novoEmpregado) {}
    public void removerEmpregado(int idEmpregado) {}
    public void obterInfoEmpregados() {}
}

public class Empregado {
    private String nome;
    private int id;
    private String cargo;
    private float salario;
    private Departamento departamento;

    public void atualizarInformacoes(String novoNome, String novoCargo, float novoSalario) {}
    public void transferirDepartamento(Departamento novoDepartamento) {}
    public Departamento obterInfoDepartamento() { return null; }
}

